hub/core/compression.py

```diff
@@ -75,6 +82,28 @@ def decompress_array(buffer: Union[bytes, memoryview], shape: Tuple[int]) -> np.
 
     try:
         img = Image.open(BytesIO(buffer))
-        return np.array(img).reshape(shape)
+        arr = np.array(img)
+        if np.prod(shape) != arr.size:
```

Comment:
could you explain what's going on here? or break it into functions? kinda difficult to follow right now